shoes-core/spec/shoes/builtin_methods_spec.rb

```diff
@@ -106,4 +78,49 @@
       end
     end
   end
+
+  %w(alert confirm).each do |type|
+    describe '##{alert}' do
+     let(:dialog) { double('dialog') }
+     let(:message) { double('message') }
+
+      before do
+        allow(Shoes::Dialog).to receive(:new).and_return(dialog)
+        expect(dialog).to receive(type).with(message).and_return(dialog)
+      end
+
+      it 'creates a new dialog' do
+        expect(app.send(type, message)).to eq(dialog)
+      end
+    end
+  end
+
+  describe '#ask' do
+    let(:dialog) { double('dialog') }
+    let(:message) { double('message') }
+    let(:args) { double('args') }
+
+    before do
+      allow(Shoes::Dialog).to receive(:new).and_return(dialog)
+      expect(dialog).to receive(:ask).with(message, args).and_return(dialog)
+    end
+
+    it 'creates a new dialog' do
+      expect(app.send(:ask, message, args)).to eq(dialog)
```

Comment:
Since we don't have any metaprogramming on the tests, what would you think of just calling `app.ask` here instead of the `send`? It makes sense in the other contexts, but I'd like to avoid it.